src/v2/components/FireList.jsx

```diff
@@ -93,21 +93,45 @@ export default function FireList(props) {
   const handleVoteForFire = useCallback((index, vote) => {
     if (index > -1 && index < nFires) {
       const fire = fires[index]
-      const {cameraID, timestamp} = fire
-      const isRealFire = vote === 'yes'
-      const isUndo = vote === 'undo'
-      const endpoint = isUndo ? '/api/undoVoteFire' : '/api/voteFire'
-
-      query.post(endpoint, {cameraID, isRealFire, timestamp}).then(() => {
-        if (isUndo) {
-          delete fire.voted
-        } else {
-          fire.voted = isRealFire
-        }
-
-        updateFires(indexOfOldFires > -1)
-      }) // TODO: Implement error handling. Allow errors to go uncaught for now
-         // asy they will be logged to the Console.
+      const promises = [fire]
+        .concat(Object.keys(fire._anglesByKey).map((k) => firesByKey[k]))
+        .reduce((a, f) => {
+          const {cameraID, timestamp} = f
+
+          // Vote only for `fire` and overlapping angles older than `fire`.
+          if (timestamp > fire.timestamp) {
+            return a
+          }
+
+          const hasVote = f.voted != null
+          const isUndo = vote === 'undo'
+
+          // Don’t try to undo nonexistant votes.
+          if (isUndo && !hasVote) {
+            return a
+          }
```

Comment:
I think it's good to have this check, but what scenario would trigger this case?

Reply:
When the user has undone some of the fires with overlapping angles by going further down the list of fires and undoing some. If they then try to undo a vote for a newer detection event further up the list, the overlapping fires that have already been undone will result in a 400 bad request with a corresponding `No votes to undo []` warning logged on the server side.

I don’t know if you want to make the undo operation unwarned in addition to being idempotent. And I should check whether trying to vote on an already voted fire results in a 400 as well.

Reply:
Confirmed: Voting on an already voted fire results in a 400 bad request with a corresponding `Multiple votes not supported [ [Object] ]` warning logged.

Reply:
Problematically, if one of the existing votes differs from the current vote request (e.g., it’s already a `'no'` and the request is a `'yes'`), then we’ll get a similar 400.

So the question is whether to honor the most recent request (let’s call that “idempotent and unwarned”?) or reject conflicting/coinciding requests (“idempotent but warned”). 

Though other interactions are possible, let’s assume that the UX model is simple: a vote on a newer detection event overrides existing votes (so `undo` undoes all votes, and a `yes` or `no` vote applies to all events uniformly)…

In the case of an undo operation, it’s not that critical a question because not being able to undo something that has already been undone or never needed undoing is fine. It still results in the expected behavior albeit with some errors—currently logged to the console, but eventually handled and either presented to the user (“true positive”) or hidden from them (“false positive”).

In the case of a recast vote, it’s more problematic. If we stick with the current (“idempotent but warned”) model, the client will have to issue an async request to undo the existing vote before applying the new one.

Reply:
Done.

  - Don’t cast the same vote twice.
  - Don’t undo nonexistent votes.
  - Undo older vote in order to overwrite it.